Client/Frontend/Browser/TabManagerStore.swift

```diff
@@ -128,19 +141,47 @@ class TabManagerStore {
         }
 
         var tabToSelect: Tab?
+        var fxHomeTab: Tab?
+        var customHomeTab: Tab?
+        
         for savedTab in savedTabs {
             // Provide an empty request to prevent a new tab from loading the home screen
             var tab = tabManager.addTab(flushToDisk: false, zombie: true, isPrivate: savedTab.isPrivate)
             tab = savedTab.configureSavedTabUsing(tab, imageStore: imageStore)
             if savedTab.isSelected {
                 tabToSelect = tab
             }
+            
+            if fxHomeTab == nil {
+                tab.isFxHomeTab ? fxHomeTab = tab : nil
+            }
+            if customHomeTab == nil {
+                tab.isCustomHomeTab ? customHomeTab = tab : nil
+            }
         }
 
         if tabToSelect == nil {
             tabToSelect = tabManager.tabs.first(where: { $0.isPrivate == false })
         }
-
+        
+        if shouldOpenHome {
+            let page = NewTabAccessors.getHomePage(prefs)
+            let customUrl = HomeButtonHomePageAccessors.getHomePage(prefs)
+            let homeUrl = URL(string: "internal://local/about/home")
+            
+            if page == .homePage, let customUrl = customUrl {
+                return customHomeTab != nil ? customHomeTab : tabManager.addTab(URLRequest(url: customUrl))
```

Comment:
I think this could be `customHomeTab ?? tabManager.addTab(URLRequest(url: customURL))`